include/swupd.h

```diff
@@ -258,6 +258,10 @@ extern int get_dirfd_path(const char *fullname);
 extern int verify_fix_path(char *targetpath, struct manifest *manifest);
 extern void set_local_download(void);
 extern struct list *files_from_bundles(struct list *bundles);
+extern int system_argv(char *const argv[]);
+extern int system_argv_fd(char *const argv[], int newstdin, int newstdout, int newstderr);
+extern int system_argv_pipe(char *const argvp1[], int stdinp1, int stderrp1,
```

Comment:
that number of parameters with the same type leads to "is the stdin the first or second after the argv?"

If you go with the macros, AFAIR the compiler will use the last value, so you can even define the defaults and let the user override, like fds inited to `-1`.